presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -435,7 +435,7 @@ private static OptionalLong fromMetastoreDistinctValuesCount(OptionalLong distin
     private static long fromMetastoreDistinctValuesCount(long distinctValuesCount, long nullsCount, long rowCount)
     {
         long nonNullsCount = rowCount - nullsCount;
-        if (nullsCount > 0) {
+        if (nullsCount > 0 && (distinctValuesCount > 1 || nonNullsCount == 0)) {
             distinctValuesCount--;
         }
```

Comment:
Add a separate IF if you like, to make it more explicit

```
// normalize distinctValuesCount in case there non null element
if (nonNullsCount > 0 && distinctValuesCount == 0){
  distinctValuesCount = 1;
}
```